transaction.go

```diff
@@ -169,52 +161,52 @@ func (t *Transaction) Rollback() error {
 // Savepoint creates a savepoint with the given name. The name is interpolated
 // directly into the SQL SAVEPOINT statement, so you must sanitize it if it is
 // derived from user input.
-func (t *Transaction) Savepoint(name string) error {
+func (t *Transaction) Savepoint(ctx context.Context, name string) error {
 	query := "savepoint " + t.dbmap.Dialect.QuoteField(name)
 	if t.dbmap.logger != nil {
 		now := time.Now()
 		defer t.dbmap.trace(now, query, nil)
 	}
-	_, err := exec(t, query)
+	_, err := t.ExecContext(ctx, query)
 	return err
 }
 
 // RollbackToSavepoint rolls back to the savepoint with the given name. The
 // name is interpolated directly into the SQL SAVEPOINT statement, so you must
 // sanitize it if it is derived from user input.
-func (t *Transaction) RollbackToSavepoint(savepoint string) error {
+func (t *Transaction) RollbackToSavepoint(ctx context.Context, savepoint string) error {
 	query := "rollback to savepoint " + t.dbmap.Dialect.QuoteField(savepoint)
 	if t.dbmap.logger != nil {
 		now := time.Now()
 		defer t.dbmap.trace(now, query, nil)
 	}
-	_, err := exec(t, query)
+	_, err := t.ExecContext(ctx, query)
```

Comment:
Shouldn't this also be in a transaction? `t.tx.ExecContext`

Reply:
This calls `gorp.Transaction.ExecContext`, which has some additional logic before delegating to `sql.Tx`:

```
	if t.dbmap.ExpandSliceArgs {
		expandSliceArgs(&query, args...)
	}

	if t.dbmap.logger != nil {
		now := time.Now()
		defer t.dbmap.trace(now, query, args...)
	}
	return maybeExpandNamedQueryAndExec(ctx, t, query, args...)
```

Reply:
I see that now, I didn't dig through the function calls far enough. Thank you.